### ISSUE TITLE
[NEW RULE] Minimize Complex Flow Constructs

### ISSUE THREAD

**Issue by anandmindfire** [mindfiredigital/eslint-plugin-hub#1]
### 1️⃣ Is your rule proposal related to a problem? Please describe

Complex control flow constructs—such as deep or unbounded recursion, unnecessary branching, or excessive nesting—can make code harder to understand, debug, and maintain. This increases the chance of bugs and makes onboarding new developers more difficult.

For example: *I’m always frustrated when reviewing code with overly recursive or convoluted control flows, especially when they lack clear termination conditions.*

### 2️⃣ Describe the solution you'd like

This rule should enforce simpler, bounded control flows. It should flag:

* Recursion without a clear and statically analyzable termination condition.
* Excessive nesting (e.g., nested `if`, `for`, or `while` blocks beyond a configurable depth).
* Use of unbounded loops or ambiguous control jumps.

✅ **Allowed**:

```js
function findFirstActiveItem(items) {
  for (let i = 0; i < items.length; i++) {
    if (items[i].status === 'active') {
      console.log('Found an active item:', items[i]);
      return items[i];
    }
  }
  console.log('No active item found.');
  return null;
}
```

❌ **Disallowed**:

```js
function traverseRecursively(node) {
  if (!node) return;
  console.log('Current node:', node.value);
  // Potentially unbounded recursion
  traverseRecursively(node.next);
}
```

### 3️⃣ Describe alternatives you've considered

* Developer code reviews and documentation comments. However, these are manual and error-prone.
* Runtime guards like counters to limit recursion depth, but static analysis via ESLint provides earlier feedback.
* Using cyclomatic complexity metrics. This could complement—but not replace—this rule's explicit control flow checks.

### 4️⃣ Example configuration

```json
{
  "rules": {
    "hub/minimize-complex-flow": ["warn", { "maxNestingDepth": 3, "allowRecursion": false }]
  }
}
```

### 5️⃣ Additional context

This rule would encourage developers to refactor recursive logic into iterative equivalents when appropriate, simplify nested blocks, and write more linear, testable functions. It's particularly useful for teams working on large-scale codebases where readability and maintainability are essential.

**Comment by biswalom22** [comment]
@anandmindfire @lakinmindfire This rule is verified with all possible test cases
